File: problems/682/solution.go
package main
func calPoints(ops []string) int {
    output, points := 0, []int{}

	for _, str := range ops {
		switch str {
		case "+":
			lastTwo := points[len(points) - 2:]
			points = append(points, lastTwo[0] + lastTwo[1])
		case "D":
			points = append(points, 2 * points[len(points) - 1])
		
		case "C":
			points = points[0:len(points) - 1]

		default:
			givenPoints, _ := strconv.Atoi(str)
			points = append(points, givenPoints)
		}
	}
	
	for _, val := range points {
		output += val
	}

	return output
}

File: problems/682/solution.js
/**
 * @param {string[]} ops
 * @return {number}
 */
var calPoints = function(ops) {
    let points = [];
    
    for (let i = 0; i < ops.length; i++) {
        switch (ops[i]) {
            case "+": {
                points.push(points[points.length - 1] + points[points.length - 2]);
                
                break;
            }
                
            case "C": {
                points.pop();
                
                break;
            }
                
            case "D": {
                points.push(points[points.length - 1] * 2);
                
                break;
            }
                
            default: {
                points.push(parseInt(ops[i]));
                
                break;
            }
        }
    }
    
    return points.reduce((cv, ac) => cv + ac, 0);
};